databricks/koalas/indexes.py

```diff
@@ -579,6 +582,8 @@ def names(self) -> List[Union[Any, Tuple]]:
     def names(self, names: List[Union[Any, Tuple]]) -> None:
         if not is_list_like(names):
             raise ValueError("Names must be a list-like")
+        if not all(is_hashable(name) for name in names):
+            raise TypeError("Index.name must be a hashable type")
```

Comment:
oh, wait. Do we have to check this in `rename`?

Reply:
Oh, Let me check.